restapi/src/main/java/io/stargate/web/docsapi/dao/DocumentDB.java

```diff
@@ -126,9 +129,23 @@ public void maybeCreateTable(String keyspaceName, String tableName) {
                   String.join(", ", allPathColumnNames)))
           .get();
 
-      dataStore
-          .query(String.format("CREATE INDEX ON \"%s\".\"%s\"(leaf)", keyspaceName, tableName))
-          .get();
+      // After creating the table, it can take up to 2 seconds for permissions cache to be updated,
+      // So retry this first index creation until it succeeds.  The other index creations should not
+      // need this behavior.
+      boolean success = false;
+      while (!success) {
+        try {
+          Uninterruptibles.sleepUninterruptibly(2, TimeUnit.SECONDS);
+          dataStore
+              .query(String.format("CREATE INDEX ON \"%s\".\"%s\"(leaf)", keyspaceName, tableName))
+              .get();
```

Comment:
consider passing a timeout duration to `get()` because it may block indefinitely.